### ISSUE TITLE
[java] CheckResultSet: False negative caused by unrelated if

### ISSUE THREAD

**Issue by UncleOwen** [pmd/pmd#7070]
**Affects PMD Version:** 7.28.0-SNAPSHOT

**Rule:** [CheckResultSet](https://docs.pmd-code.org/snapshot/pmd_rules_java_bestpractices.html#checkresultset)

**Description:** If I add the following code to CheckResultSet.xml, the test fails. Without the if(), the test succeeds.

**Code Sample demonstrating the issue:**

```java
    <test-code>
        <description>.next() is called inside unrelated if statement</description>
        <expected-problems>1</expected-problems>
        <expected-linenumbers>7</expected-linenumbers>
        <code><![CDATA[
import java.sql.*;
public class ResultSetTesting
{
    public void foo(ResultSet rs, boolean cond) throws SQLException
    {
        if (cond) {
            rs.next();
        }
    }
}
        ]]></code>
    </test-code>

```

**Expected outcome:**

PMD should report a violation at line 7, but doesn't. This is a false-negative.

**Running PMD through:** *PMD regression tests*
